rclcpp_components/cmake/rclcpp_components_add_node.cmake

```diff
@@ -0,0 +1,59 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Create a target which builds a shared library and also creates a target
+# which builds an executable
+# usage: rclcpp_components_add_node(
+#              <target> [EXCLUDE_FROM_ALL] <source1> [source2 ...])
+
+# :param EXCLUDE_FROM_ALL: exclude the target from default build target
+# :type EXCLUDE_FROM_ALL: boolean
+# :param target: the name of the shared library target
+# :type target: string
+# :param sourceN: the list of source files for executable and shared library
+# :type sourceN: list of strings
+#
+function(rclcpp_components_add_node)
+  cmake_parse_arguments(
+    ARGS
+    "EXCLUDE_FROM_ALL"
+    ""
+    ""
+    ${ARGN})
+  list(GET ARGS_UNPARSED_ARGUMENTS 0 target)
+  list(REMOVE_AT ARGS_UNPARSED_ARGUMENTS 0)
```

Comment:
Why not place `target` in the function declaration? It needs to be the first argument anyways right?

Reply:
I had proposed this hack so that the usage looks like

```
rclcpp_components_add_node(foo src/foo.cpp)
```

instead of 

```
rclcpp_components_add_node(TARGET foo src/foo.cpp)
```

Maybe's there a way to declare it without requiring it to be explicitly named by the user?

Reply:
Does it not just work if you do `function(rclcpp_components_add_node target)` instead?

Reply:
doh, yeah that should work :upside_down_face:

Reply:
No worries, CMake is crazy.